client/packages/invoices/src/InboundShipment/DetailView/modals/InboundLineEdit/TabTables.tsx

```diff
@@ -72,187 +63,300 @@ export const QuantityTableComponent = ({
-  }
-
-  columnDefinitions.push(
-    {
-      key: 'volumePerPack',
-      label: t('label.volume-per-pack'),
-      Cell: NumberInputCell,
-      cellProps: { decimalLimit: 10 },
-      width: 100,
-      accessor: ({ rowData }) => rowData?.volumePerPack,
-      setter: updateDraftLine,
-    },
-    {
-      key: 'delete',
-      width: 50,
-      Cell: ({ rowData }) => (
-        <IconButton
-          label="Delete"
-          onClick={() => removeDraftLine(rowData.id)}
-          icon={<DeleteIcon fontSize="small" />}
-        />
-      ),
-    }
-  );
-
-  const columns = useColumns<DraftInboundLine>(columnDefinitions, {}, [
+      {
+        accessorKey: 'unitsPerPack',
+        header: t('label.units-received', {
+          unit: pluralisedUnitName,
+        }),
+        size: 100,
+        defaultHideOnMobile: true,
+        accessorFn: row => {
+          return row.numberOfPacks * row.packSize;
+        },
+        Cell: ({ row, cell }) => (
+          <NumberInputCell
+            cell={cell}
+            updateFn={(value: number) => {
+              const { packSize, unitsPerPack } = row.original;
+              if (packSize !== undefined && unitsPerPack !== undefined) {
+                const unitToPacks = value / packSize;
+                const roundedPacks = Math.ceil(unitToPacks);
+                const actualUnits = roundedPacks * packSize;
+                if (roundedPacks === unitToPacks || roundedPacks === 0) {
+                  setPackRoundingMessage?.('');
+                } else {
+                  setPackRoundingMessage?.(
+                    t('messages.under-allocated', {
+                      receivedQuantity: format(NumUtils.round(value, 2)), // round the display value to 2dp
+                      quantity: format(actualUnits),
+                    })
+                  );
+                }
+                updateDraftLine({
+                  unitsPerPack: actualUnits,
+                  numberOfPacks: roundedPacks,
+                  id: row.original.id,
+                });
+                return actualUnits;
+              }
+            }}
+            disabled={isDisabled}
+            min={0}
+          />
+        ),
+      },
+      {
+        accessorKey: 'volumePerPack',
+        header: t('label.volume-per-pack'),
+        size: 100,
+        Cell: ({ row, cell }) => (
+          <NumberInputCell
+            cell={cell}
+            updateFn={(value: number) => {
+              updateDraftLine({ volumePerPack: value, id: row.original.id });
+            }}
+            disabled={isDisabled}
+            decimalLimit={10}
+          />
+        ),
+      },
+      {
+        id: 'delete',
+        header: '',
+        size: 50,
+        Cell: ({ row }) => (
+          <IconButton
+            label={t('button.delete')}
+            onClick={() => removeDraftLine(row.original.id)}
+            icon={<DeleteIcon fontSize="small" />}
+          />
+        ),
+      },
```

Comment:
Just a suggestion -- should we make this either `disabled` or completely hidden when there is only one row, since deleting the last row does nothing?